### PR TITLE
TRT-2836: Add integration tests for payload query functions

### PR DESCRIPTION
**Pull request by mstaeble**
## Summary
- Adds 53 integration tests covering all 11 functions in `pkg/db/query/payload_queries.go`
- Includes 8 new tests for the previously untested `GetTestFailuresForPayloadStream` function (6-table join, array aggregation, 14-day window, stream/arch isolation)
- Covers boundary conditions (exact reportEnd semantics for `<` vs `<=`), release/stream/architecture isolation, empty-result edge cases, and full column coverage for `GetPayloadDiff`
- Adds shared test fixture helpers in `test/integration/util/fixtures.go`

## Test plan
- [x] `gofmt` passes
- [x] `go vet ./test/integration/...` passes
- [x] `make integration` passes (190 tests, 0 failures)
- [x] `make lint` passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive integration coverage for payload queries, including filtering, ordering, aggregation, error handling, time boundaries, and empty results.
  * Added validation for payload diffs, accepted payloads, failures, OS upgrades, recent payloads, phase metrics, acceptance statistics, tags, predecessors, and pull-request details.
  * Added configurable test fixtures for release tags, pull requests, job runs, and their relationships.

### CODE REVIEW

**Comment by mstaeble** [comment]
@coderabbitai full review

File: test/integration/payload_queries_test.go [hunk 1]
```diff
@@ -0,0 +1,1289 @@
...
+	require.NoError(t, dbc.DB.Model(&outOfWindowRun).Update("url", "https://prow/run/out-of-window").Error)
+	intutil.CreateReleaseJobRun(t, dbc, outOfWindowTag.ID, outOfWindowRun.ID, job.Name, "Blocking", "Failed", "https://prow/run/out-of-window")
+	intutil.CreateProwJobRunTest(t, dbc, outOfWindowRun.ID, job.ID, test.ID, release, outOfWindowRunTime, int(v1.TestStatusFailure))
+
+	subquery := query.GetTestFailuresForPayloadStream(dbc.DB, release, stream, arch, reportEnd, "")
+	rows := scanStreamFailures(t, dbc, subquery)
+
+	require.Len(t, rows, 1, "only in-window failures should appear")
+	assert.Equal(t, 1, rows[0].FailureCount, "only the in-window failure should be counted")
+	assert.Equal(t, inWindowTag.ReleaseTag, rows[0].ReleaseTags[0])
```
[the remark this answers is not in the document]
Comment (mstaeble) [reply test/integration/payload_queries_test.go:968]: Fixed in the latest push. Added `require.Len(t, rows[0].ReleaseTags, 1)` before indexing, matching the guard pattern used in `BasicAggregation`.

File: test/integration/util/fixtures.go [hunk 2]
```diff
@@ -99,3 +110,90 @@ func CreateBug(t *testing.T, dbc *db.DB, key, status, summary string, lastChange
...
+		Architecture: arch,
+		Phase:        apitype.PayloadAccepted,
+		ReleaseTime:  releaseTime,
+	}
+	for _, opt := range opts {
+		opt(&rt)
+	}
+	require.NoError(t, dbc.DB.Create(&rt).Error, "creating ReleaseTag %q", releaseTag)
+	return rt
+}
```
[the remark this answers is not in the document]
Comment (mstaeble) [reply test/integration/util/fixtures.go:320]: Skipping this one. None of the existing fixture helpers in this file use `WithContext`, and Go's test timeout (-timeout flag) already guards against hangs. Adding context deadlines only to these helpers would be inconsistent with the rest of the file. If we want fixture-level timeouts, that should be a separate, file-wide effort.

File: test/integration/payload_queries_test.go [hunk 3]
```diff
@@ -0,0 +1,1288 @@
...
+	intutil.CreateReleaseTag(t, dbc, "4.16.0-0.nightly-2024-05-31-010000", "4.16", "nightly", "amd64",
+		time.Date(2024, 5, 31, 1, 0, 0, 0, time.UTC))
+
+	results, err := query.GetLastPayloadTags(dbc.DB, "4.16", "nightly", "amd64", reportEnd)
+	require.NoError(t, err)
+
+	require.Len(t, results, 3, "should include recent, within-window, and boundary tags")
+	assert.Equal(t, recent.ReleaseTag, results[0].ReleaseTag, "most recent should be first")
+	assert.Equal(t, within.ReleaseTag, results[1].ReleaseTag)
+	assert.Equal(t, boundary.ReleaseTag, results[2].ReleaseTag, "boundary-inclusive")
```
[the remark this answers is not in the document]
Comment (mstaeble) [reply test/integration/payload_queries_test.go:424]: The test already passes and correctly asserts zero values for the single-payload case (line 742-756). PostgreSQL returns NULL for `AVG/MIN/MAX` over a NULL duration, and Go's `database/sql` scanner maps NULL to the zero value of `int64`. The behavior is correct as-is.

Adding `COALESCE` to the production query in `payload_queries.go` would be a defensive hardening change, not a test fix. That's out of scope for this test-only PR.
